src/lib.rs

```diff
@@ -232,6 +252,30 @@ impl<T> Tree<T> {
         self.vec.extend(other_tree.vec);
         unsafe { self.get_unchecked_mut(other_tree_root_id) }
     }
+
+    /// Maps a `Tree<T>` to `Tree<U>` by applying a function to all node values,
+    /// copying over the tree's structure and node ids untouched, consuming `self`.
+    pub fn map_values<U>(self, transform: impl Fn(T) -> U) -> Tree<U> {
+        Tree {
+            vec: self
+                .vec
+                .into_iter()
+                .map(|node| node.map_value(&transform))
+                .collect(),
+        }
+    }
+
+    /// Maps a `&Tree<T>` to `Tree<U>` by applying a function to all node values,
+    /// copying over the tree's structure and node ids untouched.
+    pub fn map_value_refs<U>(&self, transform: impl Fn(&T) -> U) -> Tree<U> {
```

Comment:
Maybe we want `NodeRef<'_, T>::map -> Node<U>` as well so that we can map over an iterator created via `tree.nodes()` and collect that into a new tree?

Reply:
As discussed above, `FromIterator` semantics are too loose for us here.